From stable-bounces@linux.kernel.org Thu Jul 19 17:27:52 2007
From: Mariusz Kozlowski <m.kozlowski@tuxland.pl>
Date: Thu, 19 Jul 2007 17:27:22 -0700
Subject: fs: 9p/conv.c error path fix
To: stable@kernel.org
Cc: Latchesar Ionkov <lucho@ionkov.net>, Mariusz Kozlowski <m.kozlowski@tuxland.pl>, Eric Van Hensbergen <ericvh@ericvh.myip.org>
Message-ID: <20070719172722.1d4500f1.akpm@linux-foundation.org>


From: Mariusz Kozlowski <m.kozlowski@tuxland.pl>


When buf_check_overflow() returns != 0 we will hit kfree(ERR_PTR(err))
and it will not be happy about it.

Signed-off-by: Mariusz Kozlowski <m.kozlowski@tuxland.pl>
Cc: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 fs/9p/conv.c |    1 +
 1 file changed, 1 insertion(+)

--- a/fs/9p/conv.c
+++ b/fs/9p/conv.c
@@ -742,6 +742,7 @@ struct v9fs_fcall *v9fs_create_twrite(u3
 	if (err) {
 		kfree(fc);
 		fc = ERR_PTR(err);
+		goto error;
 	}
 
 	if (buf_check_overflow(bufp)) {
